participant/knight.py

```diff
@@ -0,0 +1,46 @@
+from __future__ import annotations
+
+
+class Knight:
+    def __init__(self, knight: dict) -> None:
+        self.name: str = knight["name"]
+        self.power: int = knight["power"]
+        self.hp: int = knight["hp"]
+        self.armour: list[dict] = knight["armour"]
+        self.weapon: dict = knight["weapon"]
+        self.potion: dict = knight["potion"]
+        self.protection: int = 0
+
+    def apply_armour(self) -> None:
+        for armour in self.armour:
+            self.protection += armour["protection"]
+
+    def apply_weapon(self) -> None:
+        self.power += self.weapon["power"]
+
+    def apply_potion(self) -> None:
+        if self.potion is not None:
+            for key, value in self.potion["effect"].items():
+                if hasattr(self, key):
+                    setattr(self, key, getattr(self, key) + value)
+
+    def apply_all_equipment(self) -> None:
+        self.apply_armour()
+        self.apply_weapon()
+        self.apply_potion()
+
+    def battle_vs(self, other_knight: Knight) -> None:
+        if not isinstance(other_knight, Knight):
+            raise TypeError(
+                f"unsupported operand type(s) for -: "
+                f"'Knights' and {type(other_knight)}"
+            )
```

Comment:
I see no point in this check because it is stated in annotation that other_knight should be Knight instance